lib/auth/auth_with_roles.go

```diff
@@ -3461,6 +3519,10 @@ func (a *ServerWithRoles) CompareAndSwapUser(ctx context.Context, new, existing
 		return trace.Wrap(err)
 	}
 
+	if err := a.checkOktaUpdateAccess(ctx, new, existing.GetName()); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Given that the intent for that check was to prevent the Okta server annexing a non-Okta user, I think that works:

| `existing` origin | Backend record origin | Access   | Consistent | Notes |
|-------------------|-----------------------|----------|------------|-------|
| Okta              | Okta                  | Approved | Yes        |       |
| Okta              | `$SOMETHING_ELSE`     | Denied   | Yes        | Update will be rejected by `CompareAndSwap` because `existing` and backend values are different      |
| `$SOMETHING_ELSE` | Okta                  | Denied   | Yes        | Update rejected by okta origin check |
| `$SOMETHING_ELSE` | `$SOMETHING_ELSE`     | Denied   | Yes        | Update rejected by okta origin check |